tests/optimize_relinearization/optimize_relinearization.mlir

```diff
@@ -0,0 +1,161 @@
+// RUN: heir-opt --optimize-relinearization %s | FileCheck %s
+
+#encoding = #lwe.polynomial_evaluation_encoding<cleartext_start=30, cleartext_bitwidth=3>
+#my_poly = #polynomial.int_polynomial<1 + x**1024>
+#ring = #polynomial.ring<coefficientType = i32, coefficientModulus = 161729713 : i32, polynomialModulus=#my_poly>
+#params = #lwe.rlwe_params<dimension=2, ring=#ring>
+#params1 = #lwe.rlwe_params<dimension=3, ring=#ring>
+
+!pt = !lwe.rlwe_plaintext<encoding=#encoding, ring=#ring, underlying_type=i3>
+!ct = !lwe.rlwe_ciphertext<encoding=#encoding, rlwe_params=#params, underlying_type=i3>
+!ct1 = !lwe.rlwe_ciphertext<encoding=#encoding, rlwe_params=#params1, underlying_type=i3>
+
+// CHECK-LABEL: func.func @two_muls_followed_by_add
+// CHECK-SAME: %[[ARG0:[^:]*]]: !lwe.rlwe_ciphertext
+// CHECK-SAME: %[[ARG1:[^:]*]]: !lwe.rlwe_ciphertext
+// CHECK-SAME: %[[ARG2:[^:]*]]: !lwe.rlwe_ciphertext
+// CHECK-SAME: %[[ARG3:[^:]*]]: !lwe.rlwe_ciphertext
+// CHECK-NEXT: %[[MUL0:.*]] = bgv.mul %[[ARG0]], %[[ARG1]]
+// CHECK-NEXT: %[[MUL1:.*]] = bgv.mul %[[ARG2]], %[[ARG3]]
+// CHECK-NEXT: %[[ADD:.*]] = bgv.add %[[MUL0]], %[[MUL1]]
+// CHECK-NEXT: %[[RELINEARIZE0:.*]] = bgv.relinearize %[[ADD]]
+// CHECK-NEXT: return %[[RELINEARIZE0]]
+
+func.func @two_muls_followed_by_add(%arg0: !ct, %arg1: !ct, %arg2: !ct, %arg3: !ct) -> !ct {
```

Comment:
The last test in this file as a rotate op.

Reply:
I guess I meant that rotated inputs that are later multiplied and relinearized work - but it really does since the basis size is the same as any other canonical ciphertext.